### PR TITLE
Remove `FlagsAttribute` from `VMState`

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -1660,25 +1680,20 @@ public bool RemoveBreakPoint(byte[] script_hash, uint position)
 
         public void StepInto()
         {
-            if (InvocationStack.Count == 0) State |= VMState.HALT;
             if (State.HasFlag(VMState.HALT) || State.HasFlag(VMState.FAULT)) return;
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs]: `==` ?

File: src/neo-vm/ExecutionEngine.cs [hunk 2]
```diff
@@ -1660,25 +1680,20 @@ public bool RemoveBreakPoint(byte[] script_hash, uint position)
...
+            ExecuteNext();
+            if (State == VMState.NONE)
+                State = VMState.BREAK;
         }
 
         public void StepOut()
         {
             State &= ~VMState.BREAK;
             int c = InvocationStack.Count;
             while (!State.HasFlag(VMState.HALT) && !State.HasFlag(VMState.FAULT) && !State.HasFlag(VMState.BREAK) && InvocationStack.Count >= c)
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs]: HasFlag should be changed to `==`

Comment (shargon) [reply src/neo-vm/ExecutionEngine.cs]: if we move set this values,  we can check if is `>=2`
```
NONE=0
BREAK=1
HALT=2
FAULT=3
```

Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs]: We should keep their value as 1, 2, 4, ... 
This make it easy to change back to flags in future if necessary.

**Comment by erikzhang** [comment]
@shargon Any other suggestion?

**Comment by shargon** [comment]
I think that just change to `equal` all `HasFlag` calls

**Comment by erikzhang** [comment]
`HasFlag` and `equal` are the same for non-flag. But `HasFlag` is more safe.

**Pull request merged**